docs/proposals/super-simple-discovery-api.md

```diff
@@ -0,0 +1,205 @@
+
+Additions include:
+
+* A cluster serial number/identifier.
+  * In an HA world, API servers may come and go and it is necessary to make sure we are talking to the same cluster as we thought we were talking to.
+* A _set_ of addresses for finding the cluster.
+  * It is implied that all of these are equivalent and that a client can try multiple until an appropriate target is found.
+  * Initially I’m proposing a flat set here.  In the future we can introduce more structure that hints to the user which addresses to try first.
+* A _set_ of root certificates to trust for this cluster.
+  * We make this a set so that root certificates can be rotated.
+* Other root certificate options
+  * Trust the system managed list of CAs.
+
+### Client caching and update
+
+**This is to be implemented in a later phase**
+
+Any client of the cluster will want to have this information.  As the configuration of the cluster changes we need the client to keep this information up to date.  It is assumed that the information here won’t drift so fast that clients won’t be able to find *some* way to connect.
+
+In exceptional circumstances it is possible that this information may be out of date and a client would be unable to connect to a cluster.  Consider the case where a user has kubectl set up and working well and then doesn't run kubectl for quite a while.  It is possible that over this time (a) the set of servers will have migrated so that all endpoints are now invalid or (b) the root certificates will have rotated so that the user can no longer trust any endpoint.
+
+## Methods
+
+Now that we know *what* we want to get to the client, the question is how.  We want to do this in as secure a way possible (as there are cryptographic keys involved) without requiring a lot of overhead in terms of information that needs to be copied around.
+
+### Method: Out of Band
+
+The simplest way to do this would be to simply put this object in a file and copy it around.  This is more overhead for the user, but it is easy to implement and lets users rely on existing systems to distribute configuration.
+
+For the `kubeadm` flow, the command line might look like:
+
+```
+kubeadm join --cluster-info-file=my-cluster.yaml
+```
+
+Note that TLS bootstrap (which establishes a way for a client to authenticate itself to the server) is a separate issue and has its own set of methods.  This command line may have a TLS bootstrap token (or config file) on the command line also.
+
+### Method: HTTPS endpoint
+
+If the ClusterInfo information is hosted in a trusted place via HTTPS you can just request it that way.  This will use the root certificates that are installed on the system.  It may or may not be appropriate based on the user's constraints.
+
+```
+kubeadm join --cluster-info-url="https://example/mycluster.yaml"
```

Comment:
As I pointed out in the SIG, we could just get away with adding a CA file path (or body) as a flag and then it can be wrapped in `NodeConfiguration` which we can load from disk via `--config` already, and we could add support for loading from HTTPS and be done with it without inventing new types and flags etc.